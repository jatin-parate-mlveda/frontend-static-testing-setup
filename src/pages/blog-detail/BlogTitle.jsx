import styles from './BlogTitle.module.scss';

export default function BlogTitle({ title }) {
  return <div className={styles.container}>{title}</div>;
}
