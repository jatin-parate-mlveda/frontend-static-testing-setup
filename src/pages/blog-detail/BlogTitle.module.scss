@import 'src/styles/variables';

.container {
  color: $red;
}
