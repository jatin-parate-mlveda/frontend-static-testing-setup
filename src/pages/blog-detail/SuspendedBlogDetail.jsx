import { lazy, Suspense } from 'react';
import { Loading } from '../../components';

const BlogDetailPage = lazy(() =>
  import('./index').then(m => ({
    default: m.BlogDetail,
  })),
);

export default function SuspendedBlogDetail(props) {
  return (
    <Suspense fallback={<Loading />}>
      <BlogDetailPage {...props} />
    </Suspense>
  );
}
