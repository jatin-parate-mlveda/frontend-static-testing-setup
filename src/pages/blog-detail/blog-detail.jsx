export const BlogDetail = () => (
  // throw Error('something');
  <h1>Blog Detail</h1>
);
