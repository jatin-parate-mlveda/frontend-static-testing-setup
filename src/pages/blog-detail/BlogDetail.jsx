import BlogTitle from './BlogTitle';

export const BlogDetail = () => (
  <div>
    <BlogTitle title='My Blog' />
    <hr />
    <div>Content area</div>
  </div>
);
