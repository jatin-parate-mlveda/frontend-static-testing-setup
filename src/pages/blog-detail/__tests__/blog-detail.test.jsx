import { render } from '@testing-library/react';
import { BlogDetail } from '../blog-detail';

test('matches snapshot', () => {
  const { container } = render(<BlogDetail />);

  expect(container.outerHTML).toMatchSnapshot();
});
