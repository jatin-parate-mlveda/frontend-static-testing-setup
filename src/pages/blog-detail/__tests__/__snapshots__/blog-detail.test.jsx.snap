// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `"<div><h1>Blog Detail</h1></div>"`;
